[2023-10-26T07:03:28.696+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_dag.load_data manual__2023-10-26T06:52:43.267376+00:00 [queued]>
[2023-10-26T07:03:28.717+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_dag.load_data manual__2023-10-26T06:52:43.267376+00:00 [queued]>
[2023-10-26T07:03:28.718+0000] {taskinstance.py:1359} INFO - Starting attempt 3 of 3
[2023-10-26T07:03:28.739+0000] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): load_data> on 2023-10-26 06:52:43.267376+00:00
[2023-10-26T07:03:28.747+0000] {standard_task_runner.py:57} INFO - Started process 3429 to run task
[2023-10-26T07:03:28.752+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'etl_dag', 'load_data', 'manual__2023-10-26T06:52:43.267376+00:00', '--job-id', '443', '--raw', '--subdir', 'DAGS_FOLDER/tugaskel8_dag.py', '--cfg-path', '/tmp/tmp364gbxg4']
[2023-10-26T07:03:28.755+0000] {standard_task_runner.py:85} INFO - Job 443: Subtask load_data
[2023-10-26T07:03:28.819+0000] {task_command.py:415} INFO - Running <TaskInstance: etl_dag.load_data manual__2023-10-26T06:52:43.267376+00:00 [running]> on host d73c1ff20372
[2023-10-26T07:03:29.142+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='kel8' AIRFLOW_CTX_DAG_ID='etl_dag' AIRFLOW_CTX_TASK_ID='load_data' AIRFLOW_CTX_EXECUTION_DATE='2023-10-26T06:52:43.267376+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-10-26T06:52:43.267376+00:00'
[2023-10-26T07:03:29.323+0000] {warnings.py:109} WARNING - /opt/***/dags/tugaskel8_dag.py:51: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  final_df.to_sql('tes_table', hook, index=False, if_exists='replace')

[2023-10-26T07:03:29.327+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/tugaskel8_dag.py", line 51, in load_data
    final_df.to_sql('tes_table', hook, index=False, if_exists='replace')
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/core/generic.py", line 2878, in to_sql
    return sql.to_sql(
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 769, in to_sql
    return pandas_sql.to_sql(
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 2378, in to_sql
    table.create()
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 903, in create
    if self.exists():
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 889, in exists
    return self.pd_sql.has_table(self.name, self.schema)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 2385, in has_table
    return len(self.execute(query, [name]).fetchall()) > 0
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 2200, in execute
    cur = self.con.cursor()
AttributeError: 'PostgresHook' object has no attribute 'cursor'
[2023-10-26T07:03:29.346+0000] {taskinstance.py:1398} INFO - Marking task as FAILED. dag_id=etl_dag, task_id=load_data, execution_date=20231026T065243, start_date=20231026T070328, end_date=20231026T070329
[2023-10-26T07:03:29.367+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 443 for task load_data ('PostgresHook' object has no attribute 'cursor'; 3429)
[2023-10-26T07:03:29.407+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2023-10-26T07:03:29.429+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
